src/SFA.DAS.EmployerAccounts/Queries/GetAccountEmployerAgreementsRemove/GetAccountEmployerAgreementsRemoveQueryHandler.cs

```diff
@@ -57,45 +60,52 @@ public async Task<GetAccountEmployerAgreementsRemoveResponse> Handle(GetAccountE
 
             if (result == null) return new GetAccountEmployerAgreementsRemoveResponse();
 
-            if (result != null)
+            foreach (var removeEmployerAgreementView in result)
             {
-                foreach (var removeEmployerAgreementView in result)
-                {
-                    removeEmployerAgreementView.HashedAgreementId =
-                        _hashingService.HashValue((long) removeEmployerAgreementView.Id);
-                    removeEmployerAgreementView.HashedAccountId = message.HashedAccountId;
+                removeEmployerAgreementView.HashedAgreementId =
+                    _hashingService.HashValue(removeEmployerAgreementView.Id);
+                removeEmployerAgreementView.HashedAccountId = message.HashedAccountId;
 
-                    if (result.Count == 1) continue;
-
-                    switch (removeEmployerAgreementView.Status)
-                    {
-                        case EmployerAgreementStatus.Pending:
-                            removeEmployerAgreementView.CanBeRemoved = true;
-                            break;
-                        case EmployerAgreementStatus.Signed:
+                var legalAgreementId = _hashingService.DecodeValue(removeEmployerAgreementView.HashedAgreementId);
 
-                            var commitmentConnectedToEntity = commitments.FirstOrDefault(c =>
-                                !string.IsNullOrEmpty(c.LegalEntityIdentifier)
-                                && c.LegalEntityIdentifier.Equals(removeEmployerAgreementView.LegalEntityCode)
-                                && c.LegalEntityOrganisationType == removeEmployerAgreementView.LegalEntitySource);
+                var agreement = await _employerAgreementRepository.GetEmployerAgreement(legalAgreementId);
 
-                            if (commitmentConnectedToEntity != null &&
-                                (commitmentConnectedToEntity.ActiveCount +
-                                 commitmentConnectedToEntity.PendingApprovalCount +
-                                 commitmentConnectedToEntity.PausedCount) != 0)
-                            {
-                                removeEmployerAgreementView.CanBeRemoved = false;
-                            }
-                            else
-                            {
-                                removeEmployerAgreementView.CanBeRemoved = true;
-                            }
+                var vacanciesSummary = await _dasRecruitService.GetVacanciesByLegalEntity(message.HashedAccountId, agreement.LegalEntityId);
```

Comment:
Is it better to move this call (and the one above) to inside of the case statement?  The data isn't used unless the status is signed.